test/logflare/lql/rules/select_rule_test.exs

```diff
@@ -0,0 +1,316 @@
+    test "creates invalid changeset for invalid path" do
+      rule = %SelectRule{}
+      params = %{path: "invalid-field-name!"}
+
+      changeset = SelectRule.changeset(rule, params)
+
+      refute changeset.valid?
+      assert changeset.errors[:path]
+    end
+
+    test "creates invalid changeset for missing path" do
+      rule = %SelectRule{}
+      params = %{}
+
+      changeset = SelectRule.changeset(rule, params)
+
+      refute changeset.valid?
+      assert changeset.errors[:path]
+    end
+
+    test "validates deeply nested field paths" do
+      rule = %SelectRule{}
+      params = %{path: "user.profile.settings.theme.colors.primary.dark_mode"}
+
+      changeset = SelectRule.changeset(rule, params)
+
+      assert changeset.valid?
+      assert changeset.changes.path == "user.profile.settings.theme.colors.primary.dark_mode"
+      assert changeset.changes.wildcard == false
+    end
+
+    test "creates changeset from existing struct" do
+      existing_rule = %SelectRule{path: "metadata.user_id", wildcard: false}
+
+      changeset = SelectRule.changeset(nil, existing_rule)
+
+      assert changeset.valid?
+      assert changeset.changes == %{}
+      assert changeset.data == existing_rule
+    end
+  end
+
+  describe "virtual_fields/0" do
+    test "returns list of virtual field names" do
+      virtual_fields = SelectRule.virtual_fields()
+
+      assert is_list(virtual_fields)
+      assert :path in virtual_fields
+      assert :wildcard in virtual_fields
+    end
+  end
+
+  describe "normalize/1" do
+    test "applies wildcard precedence when wildcard present" do
+      rules = [
+        %SelectRule{path: "field1", wildcard: false},
+        %SelectRule{path: "*", wildcard: true},
+        %SelectRule{path: "field2", wildcard: false}
+      ]
+
+      normalized = SelectRule.normalize(rules)
+
+      assert length(normalized) == 1
+      assert hd(normalized).path == "*"
+      assert hd(normalized).wildcard == true
+    end
+
+    test "deduplicates fields when no wildcard" do
+      rules = [
+        %SelectRule{path: "field1", wildcard: false},
+        %SelectRule{path: "field2", wildcard: false},
+        %SelectRule{path: "field1", wildcard: false}
+      ]
+
+      normalized = SelectRule.normalize(rules)
+
+      assert length(normalized) == 2
+      paths = Enum.map(normalized, & &1.path)
+      assert "field1" in paths
+      assert "field2" in paths
+    end
+
+    test "handles empty list" do
+      normalized = SelectRule.normalize([])
+      assert normalized == []
+    end
+
+    test "handles single wildcard rule" do
+      rules = [%SelectRule{path: "*", wildcard: true}]
+      normalized = SelectRule.normalize(rules)
+
+      assert length(normalized) == 1
+      assert hd(normalized).wildcard == true
+    end
+  end
+
+  describe "apply_wildcard_precedence/1" do
+    test "returns first wildcard rule when wildcards present" do
+      rules = [
+        %SelectRule{path: "field1", wildcard: false},
```

Comment:
what if we were to do `s:my_col.*` to select all nested cols? would wildcard be true then?